# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/groff/hotfix.patch.avr32
# Copyright (C) 2024 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

--- groff-1.23.0/src/preproc/eqn/delim.cpp.vanilla	2023-11-30 11:35:24.639264033 +0100
+++ groff-1.23.0/src/preproc/eqn/delim.cpp	2023-11-30 11:35:48.900265599 +0100
@@ -319,10 +319,10 @@
   char buf[256];
 // The format string in the sprintf below comes from a struct
 // initializer above, and is not subject to external influence.
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wformat-nonliteral"
+//#pragma GCC diagnostic push
+//#pragma GCC diagnostic ignored "-Wformat-nonliteral"
   sprintf(buf, d->chain_format, "\\\\n[" INDEX_REG "]");
-#pragma GCC diagnostic pop
+//#pragma GCC diagnostic pop
   printf(".nr " INDEX_REG " 0\n"
 	 ".de " TEMP_MACRO "\n"
 	 ".ie c%s \\{\\\n"
--- groff-1.23.0/src/preproc/pic/pic.ypp.vanilla	2023-11-30 11:36:03.153266518 +0100
+++ groff-1.23.0/src/preproc/pic/pic.ypp	2023-11-30 11:36:12.394267114 +0100
@@ -1934,11 +1934,11 @@
 	one_format += '\0';
 // We validated the format string above.  Most conversion specifiers are
 // rejected, including `n`.
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wformat-nonliteral"
+//#pragma GCC diagnostic push
+//#pragma GCC diagnostic ignored "-Wformat-nonliteral"
 	snprintf(sprintf_buf, sizeof(sprintf_buf),
 		 one_format.contents(), v[i++]);
-#pragma GCC diagnostic pop
+//#pragma GCC diagnostic pop
       }
       one_format.clear();
       result += sprintf_buf;
--- groff-1.23.0/src/include/lib.h.vanilla	2023-11-30 11:37:17.085271288 +0100
+++ groff-1.23.0/src/include/lib.h	2023-11-30 11:37:26.018271864 +0100
@@ -106,7 +106,7 @@
 
   extern char invalid_char_table[];
 
-  inline bool is_invalid_input_char(int c)
+  static inline bool is_invalid_input_char(int c)
   {
     return (c >= 0 && invalid_char_table[c]);
   }
